docs/rules/no-confusing-set-timeout.md

```diff
@@ -0,0 +1,61 @@
+# Disallow using confusing setTimeout in test (`no-confusing-set-timeout`)
+
+<!-- end auto-generated rule header -->
+
+This rule will raise a warning about confusing `jest-setTimeout` in test.
```

Comment:
Generally this section tends to describe the situation that the rule addresses rather than the specifics of the rule itself - so it shouldn't start with "this rule" since that's what the "rule details" section generally starts with.

Instead this should talk more about the situation as described in #1366 - something like "While `jest.setTimeout` can be called anywhere in a test file implying its stateful, it actually applies to the entire file and is called before any test starts..."